src/Container.php

```diff
@@ -12,66 +11,40 @@
 class Container extends Injector implements ContainerInterface
 {
 	/**
-     * @var ContainerInterface[]
+     * @var ContainerInterface
      */
-    private $containerDelegates = [];
+    private $delegate;
 
     /**
      * @var array
      */
-    private $containerCache = [];
+    private $has = [];
 
     /**
-     * @var array
-     */
-    private $delegateCache = [];
-
-    /**
-     * @inheritdoc
+     * @param ContainerInterface $delegate
      */
-    public function get($id)
+    public function setDelegate(ContainerInterface $delegate)
     {
-        if ($this->containerHas($id)) {
-            return $this->containerGet($id);
-        }
-
-        $delegate = $this->delegateWith($id);
-        if ($delegate instanceof ContainerInterface) {
-            return $delegate->get($id);
-        }
-
-        throw new NotFoundException(
-            'No entry found: ' . $id
-        );
+        $this->delegate = $delegate;
     }
 
     /**
      * @inheritdoc
      */
-    public function has($id)
-    {
-        return $this->containerHas($id)
-            || $this->delegateWith($id) instanceof ContainerInterface;
-    }
-
-    /**
-     * @param ContainerInterface $delegate
-     */
-    public function addDelegate(ContainerInterface $delegate)
+    public function get($id)
     {
-        $this->containerDelegates[] = $delegate;
-    }
+        if (!$this->has($id)) {
+            throw new NotFoundException(
+                'No entry found: ' . $id
+            );
+        }
 
-    /**
-     * @param string $id
-     * @return mixed
-     * @throws ContainerException
-     */
-    private function containerGet($id)
-    {
         try {
-            return $this->make($id);
-        } catch (InjectorException $previous) {
+            if ($this->delegate === null) {
+                return $this->make($id);
+            }
+            return $this->delegate->get($id);
```

Comment:
You shouldn't be fetching the entry from the delegate container. In `get()` you should fetch the entry from *your* container only. However dependencies of that entry should be resolved from the delegate.